### PR TITLE
Switch all string addition to using f-strings

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Now that we no longer support Python 3.5, move to the best feature 3.6+
gives us, f-strings! This will likely be a bit of a moving target if
older code gets merged.

### CODE REVIEW

**Comment by sfdye** [comment]
Let's use [pre-commit](https://github.com/asottile/pyupgrade) to enforce this

**Review by pascal-hofmann — APPROVED** [review]

**Comment by s-t-e-v-e-n-k** [comment]
@sfdye Good poiint, I'll add that in this PR, and then merge it when CI is happy again.

**Comment by s-t-e-v-e-n-k** [comment]
Wait a second, how does pyupgrade enforce this?

**Comment by sfdye** [comment]
if you use the --python36 option, it will enforce f-string i think

**Comment by henryiii** [comment]
It doesn't enforce f-strings everywhere, just where it thinks they are better. [ref](https://github.com/asottile/pyupgrade#f-strings). It would be good to use f-strings as much as possible, because they are faster. :)

**Comment by sfdye** [comment]
> It doesn't enforce f-strings _everywhere_, just where it thinks they are better. [ref](https://github.com/asottile/pyupgrade#f-strings). It would be good to use f-strings as much as possible, because they are faster. :)

That's the more accurate way to put it 😄

**Comment by s-t-e-v-e-n-k** [comment]
@sfdye pyupgrade did some far-reaching changes, but shall we merge this and cut a .1 release?

**Comment by sfdye** [comment]
The changes are non just f-string, more like applying pyupgrade on whole codebase. Could you update the PR title/description and add the commit sha our ignore list 😄

File: github/ContentFile.py [hunk 1]
```diff
@@ -58,7 +56,7 @@ def decoded_content(self):
         """
         :type: bytes
         """
-        assert self.encoding == "base64", "unsupported encoding: %s" % self.encoding
```
Comment (sfdye) [inline github/ContentFile.py:61]: need to fix this

File: github/PaginatedList.py [hunk 2]
```diff
@@ -53,12 +51,10 @@ def __getitem__(self, index):
             return self._Slice(self, index)
 
     def __iter__(self):
-        for element in self.__elements:
-            yield element
+        yield from self.__elements
         while self._couldGrow():
             newElements = self._grow()
-            for element in newElements:
-                yield element
```
Comment (sfdye) [inline github/PaginatedList.py:61]: sweet

**Comment by s-t-e-v-e-n-k** [comment]
So the problem is, if we disentangle this, we get a bunch of changes that pyuppgrade makes calling .format(). I can try again if you wish.

File: github/ContentFile.py (hunk 1 above)
Comment (s-t-e-v-e-n-k) [reply github/ContentFile.py:61]: Our assert messages are a mess anyway, but I'll put it back.

**Comment by sfdye** [comment]
> So the problem is, if we disentangle this, we get a bunch of changes that pyuppgrade makes calling .format(). I can try again if you wish.

What is the problem? I don't understand

**Comment by s-t-e-v-e-n-k** [comment]
> What is the problem? I don't understand

Objection removed -- I'll push up a new PR with pyupgrade. Given the formative changes that it will introduce, like yield form, I don't think we should add it to .git-blame-ignore-revs

**Comment by sfdye** [comment]
I feel most changes (remove unicode literal, removing shebang/convert to f-string) are inconsequential

**Pull request merged**